### PR TITLE
Fix error message for failed login when providing tenant domain

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Fix for https://github.com/Azure/azure-powershell/issues/6936

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs [hunk 1]
```diff
@@ -158,8 +158,22 @@ public bool TryRemoveContext(IAzureContext context)
                             promptAction)?.FirstOrDefault();
                         if (tenant == null || tenant.Id == null)
                         {
-                            throw new ArgumentNullException(string.Format("Could not find tenant id for provided tenant domain '{0}'. Please ensure that " +
-                                                                          "the provided service principal is found in the provided tenant domain.", tenantId));
+                            string baseMessage = string.Format("Could not find tenant id for provided tenant domain '{0}'", tenantId);
```
Comment (maddieclayton) [inline src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs]: The strings should probably be in Resources.resx.  Other than that looks good to me.

File: src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs [hunk 2]
```diff
@@ -158,8 +158,22 @@ public bool TryRemoveContext(IAzureContext context)
...
+                            else if (account.Type == AzureAccount.AccountType.User)
+                            {
+                                typeMessage = "Please ensure that the provided user is found in the provided tenant domain.";
+                            }
+                            else if (account.Type == AzureAccount.AccountType.ManagedService)
+                            {
+                                typeMessage = "Please ensure that the managed service identity found on this machine has proper permissions to the provided tenant domain.";
+                            }
+
+                            throw new ArgumentNullException(string.Format("{0} {1}", baseMessage, typeMessage));
```
Comment (MiYanni) [inline src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs]: Simplification. I'm assuming `AzureAccount.AccountType` has static strings.
```cs
var messageMap = new Dictionary<string, string>
{
    { AzureAccount.AccountType.ServicePrincipal, $"Please ensure that the provided service principal {account.Id} is found in the provided tenant domain." },
    { AzureAccount.AccountType.User,              "Please ensure that the provided user is found in the provided tenant domain." },
    { AzureAccount.AccountType.ManagedService,    "Please ensure that the managed service identity found on this machine has proper permissions to the provided tenant domain." }
};
throw new ArgumentNullException("Could not find tenant id for provided tenant domain '{tenantId}' {(messageMap.ContainsKey(account.Type) ? messageMap[account.Type] : String.Empty)}");
```

**Comment by cormacpayne** [comment]
@maddieclayton @MiYanni comments have been addressed

File: src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs [hunk 3]
```diff
@@ -158,8 +158,15 @@ public bool TryRemoveContext(IAzureContext context)
...
-                                                                          "the provided service principal is found in the provided tenant domain.", tenantId));
+                            string baseMessage = string.Format(ProfileMessages.TenantDomainNotFound, tenantId);
+                            var typeMessageMap = new Dictionary<string, string>
+                            {
+                                { AzureAccount.AccountType.ServicePrincipal, string.Format(ProfileMessages.ServicePrincipalTenantDomainNotFound, account.Id) },
+                                { AzureAccount.AccountType.User, ProfileMessages.UserTenantDomainNotFound },
+                                { AzureAccount.AccountType.ManagedService, ProfileMessages.MSITenantDomainNotFound }
+                            };
+                            string typeMessage = typeMessageMap.ContainsKey(account.Type) ? typeMessageMap[account.Type] : string.Empty;
+                            throw new ArgumentNullException(string.Format("{0} {1}", baseMessage, typeMessage));
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs]: use interpolation here:

```c#
$"{baseMessage} {typedMessage}"
```

**Pull request merged**